File: Fundamentals with JS/T10 - Functions/exercise/06. Password Validator.js
function passwordValidator(password) {
    function isLengthEnough(password) {
        return password.length >= 6 && password.length <= 10;
    }

    function isLettersAndDigits(password) {
        for (let i = 0; i < password.length; i++) {
            let currentChar = password[i].charCodeAt(0);
            if (
                !(currentChar >= 65 && currentChar <= 90) &&
                !(currentChar >= 97 && currentChar <= 122) &&
                !(currentChar >= 48 && currentChar <= 57)
            ) {
                return false;
            }
        }
        return true;
    }

    function isTwoDigits(password) {
        let counter = 0;
        for (let i = 0; i < password.length; i++) {
            let currentChar = password[i].charCodeAt(0)
            if (currentChar >= 48 && currentChar <= 57) {
                counter++;
            }
        }
        return counter >= 2 ? true : false;
    }

    let isLengthValid = isLengthEnough(password);
    let isOnlyLettersAndDigits = isLettersAndDigits(password);
    let isAtLeastwoNumbers = isTwoDigits(password);

    if (isLengthValid && isOnlyLettersAndDigits && isAtLeastwoNumbers) {
        console.log("Password is valid");
    }
    if (!isLengthValid) {
        console.log("Password must be between 6 and 10 characters");
    }
    if (!isOnlyLettersAndDigits) {
        console.log("Password must consist only of letters and digits");
    }
    if (!isAtLeastwoNumbers) {
        console.log("Password must have at least 2 digits");
    }
}
passwordValidator('logIn');
console.log('---');
passwordValidator('MyPass123');
console.log('---');
passwordValidator('Pa$s$s');

File: Fundamentals with Python/T09 - Functions/exercise/06. Password Validator.py
password = input()

def password_validator(password):
    valid_password = True
    if len(password) < 6 or len(password) > 10:
        print("Password must be between 6 and 10 characters")
        valid_password = False
    if not password.isalnum():
        print(f'Password must consist only of letters and digits')
        valid_password = False
    count_numbers = 0
    for char in password:
        if char.isnumeric():
            count_numbers += 1
    if count_numbers < 2:
        print("Password must have at least 2 digits")
        valid_password = False
    if valid_password:
        print("Password is valid")

password_validator(password)